### ISSUE TITLE
dbeaver 4.0.x - Schema (SQLID) issues in connection to z/OS DB2

### ISSUE THREAD

**Issue by MocniMax** [serge-rider/dbeaver#1374]
Hello.

With dbeaver 4.0.x new issues with current schema (SQLID) used in connection to zOS db2 rised. 
First one is that on connection to zOS db2, if used user name doesn't have its schema defined, no schema is active for connection what leads in SQL errors for any automatic data access attempt. Here is error generated when Read data in SQL console is selected on table SET100_PRIMARYINST in schema SE2:

![clipboard01](https://cloud.githubusercontent.com/assets/26117076/23464794/94886b36-fe96-11e6-989c-9c785dcb455a.jpg)

As you can see, in displayed error table name is specified as **XTKSOA1**.SET100_PRIMARYINST where **XTKSOA1** is taken as its SQLID what is actually the name of user used to connect to db2. The table real SQLID is in this case **SE2** so correct table name should be SE2.SET100_PRIMARYINST.
Same error is generated when table owner/schema/SQLID is selected.
And error is correct because generated SQL is always without SQLID:

SELECT * FROM SET100_PRIMARYINST;

Additionally, if I try to write query in SQL editor opened by Read data in SQL console menu item, SQL Assist doesn't work and editing blocks on first SQL keyword, SELECT in this case.

On dbeaver version 3.8.5 all this issues do not exist.

Please check where the problem is. Thanks.

Regards,
Mocni Max

**Comment by serge-rider** [comment]
Interesting (that this issue appeared in 4.0). Try this:
- Upgrade to 4.0.2
- Disable "Preferences->General->Editors->SQL Editor->Open separate connection for each editor" and restart DBeaver.

"Read data in SQL console" always use fully qualified table name. Do you see schemas in database navigator?

**Comment by MocniMax** [comment]
Hi Serge.

I used dbeaver 4.0.2 when I reported this issue.

The Open separate connection for each editor option in Preferences is already disabled.

In this case "Read data in SQL console" generates SELECT using only table name, not fully qualified one and that is the problem.

![clipboard01](https://cloud.githubusercontent.com/assets/26117076/23498348/ac54af90-ff27-11e6-8fc7-7f2849e3d2bf.jpg)

Yes, I see. I forgot to point that this issue happens only with connections to db2 on z/OS. When connected to Linux/Unix-AIX/Windows db2, this works fine. This is same database/table but on AIX:

![clipboard01](https://cloud.githubusercontent.com/assets/26117076/23498458/1b4ded80-ff28-11e6-94c4-30b94a8afc19.jpg)

Any idea why this happens?
